lib/dependency_validator.dart

```diff
@@ -266,15 +273,16 @@ Future<Null> run() async {
             // Remove this package, since we know they're using our executable
             ..remove(dependencyValidatorPackageName);
 
-  // Find unused packages that provide an executable. We assume those executables are used, but warn the user in case they are not.
-  final consideredUsed = unusedDependencies.intersection(packagesWithExecutables);
-  if (consideredUsed.isNotEmpty) {
-    log(Level.INFO, 'the following packages contain executables, they are assumed to be used:', consideredUsed);
-  }
-
-  // Remove deps that provide an executable, assume that the executable is used
+  // Remove deps that provide executables, those are assumed to be used
+  logIntersection(Level.INFO, 'The following packages contain executables, they are assumed to be used:',
+      unusedDependencies, packagesWithExecutables);
   unusedDependencies.removeAll(packagesWithExecutables);
 
+  // Remove deps that provide builders, those are assumed to be used
+  logIntersection(Level.INFO, 'The following packages contain builders, they are assumed to be used:',
+      unusedDependencies, packagesWithBuilders);
+  unusedDependencies.removeAll(packagesWithBuilders);
+
```

Comment:
With the above changes, we should be pretty confident that the `packagesWithBuilders` list includes packages that _are actually_ used, so I think we should reword it slightly and bump the level down so that it's only logged in verbose mode:

```suggestion
  // Remove deps that provide builders, those are assumed to be used
  logIntersection(Level.FINE, 'The following packages contain builders that are auto-applied or referenced in "build.yaml":',
      unusedDependencies, packagesWithBuilders);
  unusedDependencies.removeAll(packagesWithBuilders);

```